Fixed ingress deployed by helm for k8s versions >=1.22

Wieneo opened the pull request:
Ingress for gardener-dashboard and identity service was broken, due to the removal of the Ingress resource from API extensions/v1beta1

According to the contribution guidelines, I should tag a maintainer for this simple fix:
@holgerkoser 

**What this PR does / why we need it**:
This PR fixes helm not being able to create the necessary ingress resources on K8s clusters running version >=1.22.0
It adds additional logic to the Helm templates so that the correct apiVersion is chosen for corresponding Kubernetes versions. 

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Release note**:
<!--  Write your release note:
1. Enter your release note in the below block.
2. If no release note is required, just write "NONE" within the block.

Format of block header: <category> <target_group>
Possible values:
- category:       breaking|feature|bugfix|doc|other
- target_group:   user|operator|developer|dependency
-->
```bugfix operator
Fixed an issue preventing HELM from creating the necessary ingress resources on K8s >=v1.22.0
```

```breaking operator
Make sure that the k8s cluster, on which the dashboard will be deployed, is on version v1.19.0 or higher. Older versions are not supported anymore.
```

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/gardener/dashboard?pullRequest=1173) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/gardener/dashboard?pullRequest=1173) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/gardener/dashboard?pullRequest=1173) it.</sub>

gardener-robot-ci-1 commented:
Thank you @Wieneo for your contribution. Before I can start building your PR, a member of the organization must set the required label(s) {'reviewed/ok-to-test'}. Once started, you can check the build status in the PR checks section below.

holgerkoser commented:
@Wieneo Today we discussed your PR and we think that it would be ok to drop the support for the old api `extensions/v1beta1` completely. Is it possible to change the PR accordingly? 
We have written tests for the dashboard helm charts, which are located in the `__test__` folder. Could you please adapt these [ingress test](https://github.com/gardener/dashboard/blob/master/charts/__tests__/gardener-dashboard/ingress.spec.js) and the [ingress snapshot](https://github.com/gardener/dashboard/blob/master/charts/__tests__/gardener-dashboard/__snapshots__/ingress.spec.js.snap) to the new apiVersion. To run the tests cd into the charts folder and enter
```sh
yarn test
```

Wieneo commented:
@holgerkoser I've dropped support for the legacy API and adapted the tests to reflect that change.
~~~code
Test Suites: 10 passed, 10 total
Tests:       34 passed, 34 total
Snapshots:   33 passed, 33 total
Time:        2.231 s
Ran all test suites.
yarn test  4,77s user 0,57s system 143% cpu 3,730 total
~~~

holgerkoser commented:
@Wieneo That was quick. Thank you for also fixing the problem with the `gardenctl.legacyCommand` snapshot. Could you please add a short remark in the release note that kubernetes version <1.19.0 are not supported anymore.

holgerkoser reviewed (approved):
/lgtm

Wieneo commented:
@holgerkoser I have added a small remark to the release notes.

petersutter commented:
@Wieneo thanks, I have split it into two separate release notes with different categories (bugfix and breaking change) and updated also the texts a bit. I hope this was ok for you

Wieneo commented:
Of course that ist okay :)Am 28.03.2022 23:40 schrieb Peter Sutter ***@***.***>:
@Wieneo thanks, I have split it into two separate release notes with different categories (bugfix and breaking change) and updated also the texts a bit. I hope this was ok for you

—Reply to this email directly, view it on GitHub, or unsubscribe.You are receiving this because you were mentioned.Message ID: ***@***.***>
